extensions-contrib/druid-deltalake-extensions/src/main/java/org/apache/druid/delta/input/DeltaInputSourceReader.java

```diff
@@ -113,7 +115,14 @@ public boolean hasNext()
         if (!filteredColumnarBatchCloseableIterator.hasNext()) {
           return false; // No more batches or records to read!
         }
-        currentBatch = filteredColumnarBatchCloseableIterator.next().getRows();
+
+        io.delta.kernel.utils.CloseableIterator<FilteredColumnarBatch> filteredBatch =
+            filteredColumnarBatchCloseableIterator.next();
+
+        if (!filteredBatch.hasNext()) {
+          return false;
+        }
```

Comment:
This doesn't seem right. If the filteredBatch.hasNext returns false, then we should try the next element in  filteredColumnarBatchCloseableIterator

Reply:
There might be some utility already to flatten a list of iterators into one iterator.

Reply:
Good catch, fixed